lib/src/puppetdb-cli.cc

```diff
@@ -29,69 +29,103 @@ version()
     return PUPPETDB_CLI_VERSION_WITH_COMMIT;
 }
 
-json::JsonContainer
-parse_config() {
-    const string pdbrc_path
+string
+read_config(const string& config_path) {
+    const string expanded_config_path
     { futil::tilde_expand("~/.puppetlabs/client-tools/puppetdb.conf") };
-    const string default_config_str
-    { "{\"environments\":{\"dev\":{\"server_urls\":[\"http://127.0.0.1:8080\"]}}}" };
-    const json::JsonContainer default_config(default_config_str);
-    if (fs::exists(pdbrc_path)) {
-        const json::JsonContainer raw_config(futil::read(pdbrc_path));
-        const auto host = raw_config
-                .getWithDefault<string>("default_environment", "dev");
-        return raw_config
-                .getWithDefault<json::JsonContainer>({"environments", host},
-                                                     default_config);
-    } else {
-        return default_config;
-    }
+    return fs::exists(expanded_config_path) ?
+            futil::read(expanded_config_path) : "";
 }
 
-string
-pdb_server_url(const json::JsonContainer& config) {
-    const auto server_urls = config.get< vector<string> >("server_urls");
-    return server_urls.size() ? server_urls[0] : "http://127.0.0.1:8080";
+PuppetDBConn
+parse_config(const string& config_content) {
+    const json::JsonContainer raw_config(config_content);
```

Comment:
This is failing for me. If the user does not have a puppetdb.conf, the above read_config method returns "". When the JSON container tries to parse "" it fails and I see an "invalid json" message from [here](https://github.com/puppetlabs/leatherman/blob/master/json_container/src/json_container.cc#L37)